src/block.h

```diff
@@ -0,0 +1,163 @@
+#pragma once
+
+#include "models.h"
+
+namespace neml {
+
+/// Block update in tensor notation
+//  Update an entire block of models
+//  Input data must be in row major order (i.e. nblock is the first axes)
+//  Input and output must be as full tensors (not Mandel vectors)
+int block_evaluate(
+    std::shared_ptr<NEMLModel> model,
+    size_t nblock,
+    const double * const e_np1, const double * const e_n,
+    const double * const T_np1, const double * const T_n,
+    double t_np1, double t_n,
+    double * const s_np1, const double * const s_n,
+    double * const h_np1, const double * const h_n,
+    double * const A_np1,
+    double * const u_np1, const double * const u_n,
+    double * const p_np1, const double *  p_n);
+
+/// Constants used in the below
+#define s22 0.7071067811865476
+
+/// Fast block tensor to Mandel converter
+void t2m(const double * const tensor, double * const mandel, size_t nblock);
+
+/// Static data for t2m
```

Comment:
are you sure there is no other way to do the following? I will have a serious problem in maintaining the following static members. Whatever beside this would be better...

Reply:
They have to be static data (or else we waste time constructing them).  But I agree with your comment below on testing.